app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -602,20 +649,47 @@ class HomeFragment : Fragment() {
         )
     }
 
+    private fun updateSearchSelectorMenu(searchEngines: List<SearchEngine>) {
+        val searchEngineList = searchEngines
+            .map {
+                TextMenuCandidate(
+                    text = it.name,
+                    start = DrawableMenuIcon(
+                        drawable = it.icon.toDrawable(resources),
+                    ),
+                ) {
+                    sessionControlInteractor.onMenuItemTapped(SearchSelectorMenu.Item.SearchEngine(it))
+                }
+            }
+
+        searchSelectorMenu.menuController.submitList(searchSelectorMenu.menuItems(searchEngineList))
+    }
+
     private fun observeSearchEngineChanges() {
         consumeFlow(store) { flow ->
             flow.map { state -> state.search.selectedOrDefaultSearchEngine }
                 .ifChanged()
                 .collect { searchEngine ->
-                    if (searchEngine != null) {
-                        val iconSize =
-                            requireContext().resources.getDimensionPixelSize(R.dimen.preference_icon_drawable_size)
-                        val searchIcon =
-                            BitmapDrawable(requireContext().resources, searchEngine.icon)
-                        searchIcon.setBounds(0, 0, iconSize, iconSize)
-                        binding.searchEngineIcon.setImageDrawable(searchIcon)
-                    } else {
-                        binding.searchEngineIcon.setImageDrawable(null)
+                    searchEngine?.let {
```

Comment:
I think we should stick to using a simple if statement check for `searchEngine != null` here instead of the `let` follow by the `run`